de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/concurrent/management/JupiterClient.java

```diff
@@ -71,4 +78,30 @@ public synchronized ChecksumActivity withTimestamp(ChecksumActivity checksumActi
 
     return get(checksumActivity.getPath()).withTimestamp(checksumActivity);
   }
+
+  @Override
+  public void start() {
+    jupiterHeartbeat =
+        new ScheduledThreadPoolExecutor(1, new NamedThreadFactory("dpp-jupiter-heartbeat"));
+    jupiterHeartbeat.scheduleWithFixedDelay(
+        () -> {
+          synchronized (JupiterClient.this) {
```

Comment:
This gets a bit hard to read. I would consider moving the synchronized block to a separate method (e.g. `dispatchHeartbeats`) to improve the readability of this method. Then you could also replace the synchronized block by making the method synchronized.

This would also allow you to use a method reference instead of a lambda expression, which would  mean that all of this would fit onto one line.

I had have something like this in mind:

```Java
  @Override
  public void start() {
    jupiterHeartbeat =
        new ScheduledThreadPoolExecutor(1, new NamedThreadFactory("dpp-jupiter-heartbeat"));

    jupiterHeartbeat.scheduleWithFixedDelay(this::dispatchHeartbeats, 1, 1, TimeUnit.MINUTES);
  }

  /** Dispatches an activity for every open document functioning as a heartbeat. */
  private synchronized void dispatchHeartbeats() {
    clientDocs.forEach(
        (path, jupiter) -> {
          JupiterActivity heartbeat =
              jupiter.generateJupiterActivity(
                  new TimestampOperation(), sarosSession.getLocalUser(), path);
          fireActivity(heartbeat);
        });
  }
```